Updated return parameter of YouTubeSearchTool

ansilmbabl opened the pull request:
**Description:** 
changed return parameter of YouTubeSearchTool
 

1. changed the returning links of youtube videos by adding prefix "https://www.youtube.com", now this will return the exact links to the videos
2. updated the returning type from 'string' to 'list', which will be more suited for further processings

 **Issue:** 
Fixes #10742

 **Dependencies:** 
None


<!-- Thank you for contributing to LangChain!

Replace this entire comment with:
  - **Description:** changed return parameter of YouTubeSearchTool
  - **Issue:** the issue # it fixes (if applicable),
  - **Dependencies:** None
  - **Tag maintainer:** for a quicker response, tag the relevant maintainer (see below),
  - **Twitter handle:** we announce bigger features on Twitter. If your PR gets announced, and you'd like a mention, we'll gladly shout you out!

Please make sure your PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` to check this locally.

See contribution guidelines for more information on how to write/run tests, lint, etc: 
https://github.com/hwchase17/langchain/blob/master/.github/CONTRIBUTING.md

If you're adding a new integration, please include:
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/extras` directory.

If no one reviews your PR within a few days, please @-mention one of @baskaryan, @eyurtsev, @hwchase17.
 -->

ansilmbabl committed:
Updated return parameter of YouTubeSearchTool

* changed the returning links of youtube videos by adding prefix "https://www.youtube.com", now this will return the exact links to the videos
* updated the returning type from 'string' to 'list', which will be more
  suited for further processings

eyurtsev commented on libs/langchain/langchain/tools/youtube/search.py:
```diff
@@ -32,8 +32,8 @@ def _search(self, person: str, num_results: int) -> str:
-        url_suffix_list = [video["url_suffix"] for video in data["videos"]]
-        return str(url_suffix_list)
+        url_suffix_list = ["https://www.youtube.com" + video["url_suffix"] for video in data["videos"]]
+        return url_suffix_list
```
1) Is the base URL available in the raw results 
https://github.com/joetats/youtube_search/blob/master/youtube_search/__init__.py#L39

2) _search returns a string as it's an input into an LLM, so we need to keep a cast to `str`

ansilmbabl replied:
1. No, the base URL is not available in the raw results; we only have the URL suffix starting with `"/watch?v="` in the raw data. You can refer to the JSON file returned [here](https://github.com/ansilmbabl/dump/blob/fa83f703428969637835bb54159c7a1088af714b/Open-source%20/Langchain/YouTubeSearchTool/video_data.json#L69) for a detailed view of the data structure.

2. Why we need to return the YouTube links to an LLM when it can't analyze the content within those links? Our primary goal is to obtain the video links themselves. Please correct me if my understanding is incorrect :)

hwchase17 replied:
@eyurtsev tools can actually return `Any` - @ansilmbabl if you want to add back in that change in a future PR we can probably accept it

ansilmbabl replied:
thank you, I will create a PR then

ansilmbabl committed:
Update search.py

added youtube base url "https://www.youtube.com/"
return parameter converted back to string type

ansilmbabl committed:
Merge branch 'langchain-ai:master' into ansilmbabl/YouTubeSearchTool

hwchase17 committed:
cr
